### PR TITLE
Add support for Solr 9.* replication-lag check add basic auth

### PR DESCRIPTION
**Pull request by danieltoader**
## Pull Request Checklist

**Is this in reference to an existing issue?**
Replication lag no longer works after version 9.0: https://solr.apache.org/guide/solr/latest/upgrade-notes/major-changes-in-solr-9.html

`If you rely on metrics, alerts, or monitors on Solr KPIs that use the "master" or "slave" terminology, please update your system for those metrics to now show up with "leader" and "follower" terminology.`

#### General

- [x] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by danieltoader** [comment]
@majormoses, @jaredledvina any change of getting this merged and released as v1.2.0?

Sorry for the tags, I saw you did the review and merge on an old PR.

I tested it with both Solr 8 and Solr 9, with and without auth params, I can remove them if it might help the PR get merged faster.

Thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ This CHANGELOG follows the format listed [here](https://github.com/sensu-plugins
 
 ## [Unreleased]
 
+## [1.2.0] - 2023-10-14
+### Changed
+- Changed solr replication lag check to also support solr 9.* (@danieltoader)
```
Comment (majormoses) [inline CHANGELOG.md:10]:
```suggestion
- Added backwards compatible support for solr 9.x in `check-solr-replication-lag` by checking both `(follower|slave)` terms in  API response (@danieltoader)
- Added authentication in `check-solr-replication-lag` (@danieltoader)
```

**Review by majormoses — APPROVED** [review]
LGTM

@danieltoader, thanks for your contribution. We typically have the versions bumped by the maintainers, but that's fine in this case.

**Comment by majormoses** [comment]
Released:
- https://github.com/sensu-plugins/sensu-plugins-solr/releases/tag/1.2.0
- https://rubygems.org/gems/sensu-plugins-solr/versions/1.2.0

**Pull request merged**